### PR TITLE
SCE Room Speakers

### PR DESCRIPTION
**Pull request by jangomax**

### CODE REVIEW

**Review by pnavab — APPROVED** [review]
looks suspicious but if it works it works

**Review by EtienneDevictor — APPROVED** [review]
Looks good

File: api/peripheral_api/routes/Speaker.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  FORBIDDEN,
+  NOT_FOUND,
+} = require('../../util/constants').STATUS_CODES;
+const logger = require('../../util/logger');
+
+router.post('/stream', async (req, res) => {
+  /*
+     * How these work with Quasar:
+     * https://github.com/SCE-Development/Quasar/wiki/How-do-Health-Checks-Work%3F
+     */
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:22]: remove this and the other comment in all the other handlers, this isnt quasar

File: api/peripheral_api/routes/Speaker.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+const {
+  OK,
+  BAD_REQUEST,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND,
+} = require('../../util/constants').STATUS_CODES;
+const logger = require('../../util/logger');
+
+router.post('/stream', async (req, res) => {
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:18]: all of these functions should be expecting a token other wise anyone can play music on the speakers over the public internet

File: api/peripheral_api/routes/Speaker.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+     * https://github.com/SCE-Development/Quasar/wiki/How-do-Health-Checks-Work%3F
+     */
+  await axios
+    .post(`http://host.docker.internal:18000/stream/?url=${encodeURIComponent(req.query.url)}`)
+    .then(() => {
+      return res.sendStatus(OK);
+    })
+    .catch((err) => {
+      logger.error('Printer SSH tunnel is down: ', err);
+      return res.sendStatus(500);
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:30]: add a status code for 500 like `INTERNAL_SERVER_ERROR` in util/constants so we can do

```suggestion
      return res.sendStatus(INTERNAL_SERVER_ERROR);
```

File: api/peripheral_api/routes/Speaker.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     * How these work with Quasar:
+     * https://github.com/SCE-Development/Quasar/wiki/How-do-Health-Checks-Work%3F
+     */
+  await axios
+    .post(`http://host.docker.internal:18000/stream/?url=${encodeURIComponent(req.query.url)}`)
+    .then(() => {
+      return res.sendStatus(OK);
+    })
+    .catch((err) => {
+      logger.error('Printer SSH tunnel is down: ', err);
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:29]: is it the printer ssh tunnel thats down or something else? id log something like
```suggestion
      logger.error('/Speaker/stream had an error: ', err);
```

and then do it for all the other handlers in here

File: docker/Dockerfile [hunk 5]
```diff
@@ -33,4 +33,6 @@ COPY --from=builder /frontend/build /usr/share/nginx/html
 
 EXPOSE 3000
 
+EXPOSE 18000
```
Comment (evanugarte) [inline docker/Dockerfile:36]: we dont need this

File: src/config/config.example.json [hunk 6]
```diff
@@ -1,5 +1,6 @@
 {
   "sitekey": "super secret key",
   "TINYMCE_API_KEY": "XXXXXXXXXXXXXXXXXXXXX",
-  "GOOGLE_API_CLIENT_ID": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.apps.googleusercontent.com"
+  "GOOGLE_API_CLIENT_ID": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX.apps.googleusercontent.com",
```
Comment (evanugarte) [inline src/config/config.example.json:4]: i dont think we need changes in this file?

File: docker-compose.yml [hunk 7]
```diff
@@ -73,11 +73,15 @@ services:
...
     restart: 'on-failure'
+    environment:
+      - PERIPHERAL_API_URL=http://sce-peripheral-api:8081/peripheralapi
     stdin_open: true
     links:
       - mongodb
     depends_on:
       - mongodb
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (evanugarte) [inline docker-compose.yml:84]: undo the changes in this file, its wrong anyway, see https://github.com/SCE-Development/Clark/blob/9f0aee7de664b9de37b10b7c0228a2add453f60d/docker/Dockerfile#L16

File: src/APIFunctions/Speaker.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+import axios from 'axios';
+import { ApiResponse } from './ApiResponses';
+
+let PERIPHERAL_API_URL = process.env.REACT_APP_PERIPHERAL_API_URL
+  || 'http://localhost:8081/peripheralapi';
+
+
+console.debug(PERIPHERAL_API_URL);
```
Comment (evanugarte) [inline src/APIFunctions/Speaker.js:8]: remove all console statements in this file, every api should also be sending a token

File: docker/Dockerfile (hunk 5 above)
Comment (akshtshrma24) [reply docker/Dockerfile:36]: ya we dont ill remove :>

File: docker-compose.yml (hunk 7 above)
Comment (akshtshrma24) [reply docker-compose.yml:84]: yes

File: api/peripheral_api/routes/Speaker.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+} = require('../../util/constants').STATUS_CODES;
+const logger = require('../../util/logger');
+
+router.post('/stream', async (req, res) => {
+  if (!checkIfTokenSent(req)) {
+    logger.warn('/updateSignText was requested without a token');
+    return res.sendStatus(UNAUTHORIZED);
+  }
+  await axios
+    .post(`http://host.docker.internal:18000/stream/?url=${encodeURIComponent(req.query.url)}`)
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:25]: make this a global variable at the top of the file like https://github.com/SCE-Development/Clark/blob/9f0aee7de664b9de37b10b7c0228a2add453f60d/api/peripheral_api/routes/Cleezy.js#L19

the actual url should be passed in by docker-compose.yml like we do for cleezy, see

https://github.com/SCE-Development/Clark/blob/9f0aee7de664b9de37b10b7c0228a2add453f60d/docker-compose.yml#L24

File: api/peripheral_api/routes/Speaker.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND,
+  SERVER_ERROR,
+} = require('../../util/constants').STATUS_CODES;
+const logger = require('../../util/logger');
+
+router.post('/stream', async (req, res) => {
+  if (!checkIfTokenSent(req)) {
+    logger.warn('/updateSignText was requested without a token');
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:21]: `updateSignText` was requested without a token? fix this and the other  logs in this file

File: api/peripheral_api/routes/Speaker.js [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    .then(() => {
+      return res.sendStatus(OK);
+    })
+    .catch((err) => {
+      logger.error('/Speaker/stream had an error: ', err);
+      return res.sendStatus(SERVER_ERROR);
+    });
+});
+
+router.get('/queued', async (req, res) => {
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:67]: we dont use this in the frontend, lets remove this + api function and add it back when in another

File: api/peripheral_api/util/Speaker.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
+const axios = require('axios');
```
Comment (evanugarte) [inline api/peripheral_api/util/Speaker.js:1]: remove this file since we arent using this yet

File: src/Pages/Speaker/Speaker.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+  };
+
+  // useEffect(() => {
+  //   getQueued()
+  //     .then(queued => {
+  //       console.debug("HERE")
+  //       console.debug(queued)
+  //       setQueue(queued);
+  //     });
+  // }, []);
```
Comment (evanugarte) [inline src/Pages/Speaker/Speaker.js:24]: remove this and the commented out code, we will use it in another pr

**Review by evanugarte — COMMENTED** [review]
when you finish these comments lets do unit test

File: src/Pages/Speaker/Speaker.js [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+            <Input placeholder='Enter YouTube Link' onChange=
+              {(e) => setUrl(e.target.value)}></Input>
+          </Col>
+          <Col>
+            <Button onClick={playSong}>Play</Button>
+          </Col>
+        </Row>
+        <Col>
+          <Button onClick={skipSong}>Skip</Button>
+          <Button style={{marginLeft : '1rem'}} onClick={pauseSong}>Puase</Button>
```
Comment (evanugarte) [inline src/Pages/Speaker/Speaker.js:51]:
```suggestion
          <Button style={{marginLeft : '1rem'}} onClick={pauseSong}>Pause</Button>
```

File: api/peripheral_api/routes/Speaker.js [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+  checkIfTokenSent,
+} = require('../../util/token-verification');
+const {
+  OK,
+  UNAUTHORIZED,
+  SERVER_ERROR,
+} = require('../../util/constants').STATUS_CODES;
+const logger = require('../../util/logger');
+let SPEAKER_URL = process.env.SPEAKER_URL
+  || 'http://localhost:8000';
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:16]: what port does the speaker api listen on without the ssh tunnel?

File: api/peripheral_api/routes/Speaker.js [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+router.post('/stream', async (req, res) => {
+  logger.error('IT IS HERE');
```
Comment (evanugarte) [inline api/peripheral_api/routes/Speaker.js:19]: remove random console logs

File: src/Pages/Speaker/speaker.css [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+  .button-container > button {
+    width: 100%;
+    margin: 0.5rem;
+  }
+
+  label {
+    font-size: large;
+  }
+
+  input {
```
Comment (evanugarte) [inline src/Pages/Speaker/speaker.css:36]: see https://github.com/SCE-Development/Clark/pull/1256

**Review by evanugarte — APPROVED** [review]
